zsh-history-substring-search.zsh

```diff
@@ -247,11 +247,8 @@ _history-substring-search-begin() {
     # Escape and join query parts with wildcard character '*' as seperator
     # `(j:CHAR:)` join array to string with CHAR as seperator
     #
-    if [[ -n $HISTORY_SUBSTRING_SEARCH_PREFIX ]]; then
-        local search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
-    else
-        local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
-    fi
+    search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    test -z "$HISTORY_SUBSTRING_SEARCH_PREFIX" && search_pattern="*"$search_pattern
```

Comment:
It's odd that the variable expansion is unquoted here.  Please put it inside the quotes:

    search_pattern="*$search_pattern"

Reply:
Fixed.